docs/Grammar.md

```diff
@@ -0,0 +1,88 @@
+# Pact Grammar
+
+Here ye lies the pact bnf, based on [the current spec](https://pact-language.readthedocs.io/en/stable/pact-reference.html).
+
+Note: happy is more efficient in parsing left-recursive s-expressions, so the grammar is being stated that way
+(see: https://www.haskell.org/happy/doc/html/sec-sequences.html).
+
+```ebnf
+(* A program is a collection of modules and expressions *)
+program        ::= top_level_list
+top_level_list ::=    module top_level_list | 
+                        expr top_level_list | 
+                   interface top_level_list | ε
+
+module ::= '(' 'module' ident kset doc_or_meta decl_list ')'
```

Comment:
For an example of a complex module with an interface, see here: 

https://github.com/kadena-io/chainweb-node/blob/master/pact/coin-contract/coin.pact
https://github.com/kadena-io/chainweb-node/blob/master/pact/coin-contract/coin-sig.pact

One thing that we'll need to spec are `defpact`s, which are multistep coroutines - I'll commit the code tomorrow modeling their use, but there's an example above in `coin.pact`.